numpy/distutils/command/build.py

```diff
@@ -27,6 +33,9 @@ def initialize_options(self):
         old_build.initialize_options(self)
         self.fcompiler = None
         self.warn_error = False
+        self.cpu_baseline = "min"
+        self.cpu_dispatch = "max -xop -fma4" # drop AMD legacy features by default
+        self.disable_optimization = False
```

Comment:
These flags are added 3 times: in `build.py`, `build_clib.py` and `build_ext.py`. There should be a way to refactor them to only be defined once. Although I see this is the way it is done for `fcompiler=`, which is also added 3 times. So maybe there is no way around it? @rgommers any thoughts? In any case, the defaults should always be the same: here they are changed. Is that on purpose?

The extensive comment at the top of the pull request should be part of some documentation for these options. Either in a docstring or as part of the `doc/source/*.rst` documentation.

Reply:
I can only enable it within 'build.py' but the I idea behind adding three of them is to allow separate configuring for C libs building and extensions building.

for example, the user can disable the optimizations for `npyrandom`,  `npymath`,  and at the same allow it only for NumPy extensions.
```Bash
python3 setup.py build_clib --disable_optimization build install
````